tests/integration/nm/ovs_test.py

```diff
@@ -132,115 +132,122 @@ def test_bridge_with_bond_and_two_slaves(
     }
     bridge_desired_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE].append(port_state)
 
-    with _bridge_interface(bridge_desired_state):
-        bridge_current_state = _get_bridge_current_state()
+    with _bridge_interface(nm_plugin.client, bridge_desired_state):
+        bridge_current_state = _get_bridge_current_state(nm_plugin)
         assert bridge_desired_state == bridge_current_state
 
-    assert not _get_bridge_current_state()
+    assert not _get_bridge_current_state(nm_plugin)
 
 
 @contextmanager
-def _bridge_interface(state):
+def _bridge_interface(client, state):
     try:
-        _create_bridge(state)
+        _create_bridge(client, state)
         yield
     finally:
-        _delete_iface(BRIDGE0)
+        _delete_iface(client, BRIDGE0)
         for p in state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE]:
-            _delete_iface(nm.ovs.PORT_PROFILE_PREFIX + p[OB.Port.NAME])
-            _delete_iface(p[OB.Port.NAME])
+            _delete_iface(client, nm.ovs.PORT_PROFILE_PREFIX + p[OB.Port.NAME])
+            _delete_iface(client, p[OB.Port.NAME])
 
 
-@nmclient_context
-def _get_bridge_current_state():
+def _get_bridge_current_state(nm_plugin):
+    nm_plugin.refresh_content()
```

Comment:
Why is the `nm_plugin.refresh_content()` needed here? Thanks!

Reply:
Sine we are sharing single `NM.Client` in the single call of `libnmstate.apply()`, the data for query/verification will not be updated if don't iterate the main_context.
For the old code, we remove old `NM.Client` and create new one which refreshed the data for query/verification.

Reply:
Oh. Thanks for explaining.